crates/datastore/src/locking_tx_datastore/datastore.rs

```diff
@@ -715,17 +715,16 @@ impl TxMetrics {
                 let mut table_stats =
                     <HashMap<_, _, _> as HashCollectionExt>::with_capacity(tx_data.num_tables_affected());
                 for (table_id, _) in tx_data.table_ids_and_names() {
-                    let table = committed_state
-                        .get_table(table_id)
-                        .expect("should have a table in committed state for one in tx data");
-                    table_stats.insert(
-                        table_id,
-                        TableStats {
-                            row_count: table.row_count,
-                            bytes_occupied_overestimate: table.bytes_occupied_overestimate(),
-                            num_indices: table.num_indices(),
-                        },
-                    );
+                    committed_state.get_table(table_id).and_then(|table| {
+                        table_stats.insert(
+                            table_id,
+                            TableStats {
+                                row_count: table.row_count,
+                                bytes_occupied_overestimate: table.bytes_occupied_overestimate(),
+                                num_indices: table.num_indices(),
+                            },
+                        )
+                    });
```

Comment:
This reminds me: do we need to clean up any metrics when a table gets deleted? If there are any metrics that have a table ID in the `with_label_values`, we'll want to call `remove_label_values` on them at some point, probably within commit.